misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLease.kt

```diff
@@ -195,8 +196,13 @@ internal class ZkLease(
   /** @return true if we should hold the lease per the cluster membership */
   private fun shouldHoldLease(): Boolean {
     val clusterSnapshot = manager.cluster.snapshot
-    val desiredLeaseOwner = clusterSnapshot.resourceMapper[leaseResourceName]
-    return desiredLeaseOwner.name == clusterSnapshot.self.name
+    val desiredLeaseOwner = try {
+      clusterSnapshot.resourceMapper[leaseResourceName]
+    } catch (e: NoMembersAvailableException) {
```

Comment:
can you update the javadoc for `ClusterResourceMapper` to say it throws the exception?